scripts/avalon8/avalon871-pmu-test.py

```diff
@@ -0,0 +1,422 @@
+        return False
+#AD1~AD5
+    for index in range(len(pmu_state_name)):
+        a = int(binascii.hexlify(res[(index * 2 + 6):(index * 2 + 8)]), 16)
+        if (index < 1):
+            print(pmu_state_name[index] + '%d' %a + '(%f' %convert_to_temp(a) + 'C)')
+        elif (index == 1):
+            print(pmu_state_name[index] + '%d' %a + '(%f' %convert_to_vcc(convert_to_vin(a)) + 'V)')
+        else:
+            print(pmu_state_name[index] + '%d' %a + '(%f' %convert_to_vcore(convert_to_vin(a)) + 'V)')
+#PG1~PG3
+    a = binascii.hexlify(res[16:18])
+    pmu_pg_state_key = pmu_pg_state.keys()
+    for index in range(len(pmu_pg_state_key)):
+        if a == pmu_pg_state_key[index]:
+            print("PG1:    " + pmu_pg_state.get(pmu_pg_state_key[index]))
+    a = binascii.hexlify(res[18:20])
+    pmu_pg_state_key = pmu_pg_state.keys()
+    for index in range(len(pmu_pg_state_key)):
+        if a == pmu_pg_state_key[index]:
+            print("PG2:    " + pmu_pg_state.get(pmu_pg_state_key[index]))
+    a = binascii.hexlify(res[20:22])
+    pmu_pg_state_key = pmu_pg_state.keys()
+    for index in range(len(pmu_pg_state_key)):
+        if a == pmu_pg_state_key[index]:
+            print("PG3:    " + pmu_pg_state.get(pmu_pg_state_key[index]))
+#LED1~LED3
+    a = binascii.hexlify(res[22:24])
+    pmu_led_state_key = pmu_led_state.keys()
+    for index in range(len(pmu_led_state_key)):
+        if a == pmu_led_state_key[index]:
+            print("LED1:   " + pmu_led_state.get(pmu_led_state_key[index]))
+    a = binascii.hexlify(res[24:26])
+    pmu_led_state_key = pmu_led_state.keys()
+    for index in range(len(pmu_led_state_key)):
+        if a == pmu_led_state_key[index]:
+            print("LED2:   " + pmu_led_state.get(pmu_led_state_key[index]))
+    a = binascii.hexlify(res[26:28])
+    pmu_led_state_key = pmu_led_state.keys()
+    for index in range(len(pmu_led_state_key)):
+        if a == pmu_led_state_key[index]:
+            print("LED3:   " + pmu_led_state.get(pmu_led_state_key[index]))
+#SET_NTC_MAX
+    a = int(binascii.hexlify(res[28:30]), 16)
+    print('NTC_MAX:    ' + '%d' %a + '(%f' %convert_to_temp(a) + 'C)')
+    return True
+
+def test_polling():
+    while (True):
+        h = raw_input("Please input(1-4), h for help:")
+        if (h == 'h') or (h == 'H'):
+            show_help()
+        elif (h == 'q') or (h == 'Q'):
+            sys.exit(0)
+        elif h == '1':
+            detect_version()
+        elif h == '2':
+            vol = raw_input("Please input the voltage:")
+            set_vol_value(vol)
+        elif h == '3':
+            led = raw_input("Please input the led state:")
+            set_led_state(led)
+        elif h == '4':
+            if (get_state() == False):
+                sys.exit(0)
+        else:
+            show_help()
+
+if __name__ == '__main__':
+    while (True):
+        if options.is_rig == '0':
+            set_led_state("000000010000020000")
```

Comment:
Is there any docs about the magic string ?
Or could you change it to a function ?
It will be more clear.

Thanks.

Reply:
Do you need documentation on the format of the incoming parameter string?

Thanks.